File: proposed-projects/round-trip-experiment/reverse/20260909T011452Z_9f286468/20260910T045943Z_8c3d5bc1/ported_implementation/src/utils/errors/get-input-as-string.ts
import type { ValidInput } from '../../grammar-graph/types';

export const getInputAsString = (src: ValidInput): string => {
  if (typeof src === 'string') {
    return src;
  }
  const codePoints = Array.isArray(src) ? src : [src];
  return codePoints.map((cp) => String.fromCodePoint(cp)).join('');
};

File: packages/gbnf-experiment/data/20260909T185402Z_14873f98/ported_implementation/gbnf/utils/errors/get_input_as_string.py
from typing import Union

ValidInput = Union[str, int, list]


def get_input_as_string(src: ValidInput) -> str:
    if isinstance(src, str):
        return src
    code_points = src if isinstance(src, list) else [src]
    return "".join(chr(cp) for cp in code_points)
